data/battle_scripts_1.s

```diff
@@ -439,6 +439,97 @@ gBattleScriptsForMoveEffects::
 	.4byte BattleScript_EffectHit                     @ EFFECT_GIGATON_HAMMER
 	.4byte BattleScript_EffectSaltCure                @ EFFECT_SALT_CURE
 
+BattleScript_EffectPledge::
+	attackcanceler
+	accuracycheck BattleScript_PrintMoveMissed, ACC_CURR_MOVE
```

Comment:
There shouldn't be an accuracy check for the first Pokemon (the one that waits for its partner).